htdocs/comm/propal/class/propal.class.php

```diff
@@ -2629,7 +2629,10 @@ public function closeProposal($user, $status, $note = '', $notrigger = 0)
 		$newprivatenote = dol_concatdesc($this->note_private, $note);
 
 		$sql  = "UPDATE ".MAIN_DB_PREFIX."propal";
-		$sql .= " SET fk_statut = ".((int) $status).", note_private = '".$this->db->escape($newprivatenote)."', date_signature='".$this->db->idate($now)."', fk_user_signature=".$user->id;
+		$sql .= " SET fk_statut = ".((int) $status).", note_private = '".$this->db->escape($newprivatenote)."', date_signature='".$this->db->idate($now)."', fk_user_signature=".((int) $user->id);
+		if ($status == self::STATUS_NOTSIGNED) {
+			$sql .= ", date_cloture='".$this->db->idate($now)."', fk_user_cloture=".((int) $user->id);
```

Comment:
The field date_cloture and fk_user_cloture is when status of proposal is set to 
STATUS_BILLED = 4;    // that means STATUS_CLOSE. will be renamed later

Reply:
Yes, better to set no date (we have no date to store the "refuse" status).